pages/blog/posts/understanding-lexical-dynamic-scopes.md

```diff
@@ -0,0 +1,244 @@
+---
+title: "Understanding JSON Schema Lexical and Dynamic Scopes"
+date: "2024-01-19"
+tags:
+  - Fundamentals
+  - Scopes
+type: Engineering
+cover: /img/posts/2024/understanding-lexical-dynamic-scopes/cover.webp
+authors:
+  - name: Juan Cruz Viotti
+    photo: /img/avatars/jviotti.webp
+    link: https://www.jviotti.com
+    byline: O'Reilly author, founder of Intelligence.AI and Sourcemeta
+excerpt: "A deep dive on lexical and dynamic scopes, essential for understanding more advanced topics like dynamic referencing"
+---
+
+Most of the keywords defined by the JSON Schema organization can be either
+evaluated on their own, or by only considering the values of their adjacent
+keywords. For example, the `type` keyword is independent of any other keywords,
+while the `minContains` and `maxContains` keywords depend on the `contains`
+keyword defined in the same schema object.
+
+However, there is a small set of keywords whose evaluation depend on the
```

Comment:
```suggestion
However, there are a small set of keywords whose evaluation depend on the
```
I think this is common to both British and US English -- the rule (which I don't know by name by heart, I had to double check what it is, just to make it sound right) has to do with when you're using a collective noun (set) but really referring to each of its members individually, which is the case here.

Reply:
It doesn't sound quite right to me (in American English) as I say, I think this fits under cases where you're referring to each individual keyword, in which case a noun like `set` still gets plural in American English -- but it doesn't matter much, I'm sure leaving it as-is is fine too so feel free to ignore.